### ISSUE TITLE
New uncovered (edge-?)case : parse_string ValueError

### ISSUE THREAD

**Issue by jeanbaptistemora** [dtcooper/python-fitparse#75]
Hello everyone,

I wanted to report a small bug - or rather an uncovered case - in the get_messages() function, and see if anyone else sees the same thing.

We have a system in production extracting ~800 fit files per day and since 15 days we have seen ~ 50 (increasing by the day) fit files with an exception : 

```
File "/xxxx/garmin.py", line 224, in build_sessions
    for message in fit.get_messages(as_dict=True):
  File "/usr/local/lib/python3.5/dist-packages/fitparse/base.py", line 450, in get_messages
    message = self._parse_message()
  File "/usr/local/lib/python3.5/dist-packages/fitparse/base.py", line 160, in _parse_message
    message = self._parse_data_message(header)
  File "/usr/local/lib/python3.5/dist-packages/fitparse/base.py", line 320, in _parse_data_message
    raw_values = self._parse_raw_values_from_data_message(def_mesg)
  File "/usr/local/lib/python3.5/dist-packages/fitparse/base.py", line 271, in _parse_raw_values_from_data_message
    raw_value = base_type.parse(raw_value)
  File "/usr/local/lib/python3.5/dist-packages/fitparse/records.py", line 336, in parse_string
    end = string.index(0x00)
ValueError: substring not found
```

I'm not a specialist but I've tried to debug this and started by outputting the raw_value for these files, here are some of them : 

b'IM G 33 NTAP'
b'S0E2 - 1h + 4*10'
b'CAP Fartleck VMA'
b'Basse Intensit\xc3\xa9'
b'RF5 or Cross-Tra'

I think it's user inputs, and they are confusing the parse_string() function because for some reason don't contain the 0x00 substring.
It must be a new functionality from Garmin that's incorporated in the fit files for the people who have updated their hardware ?
Furthemore the mesg_type.name == 'developer_data_id' (not sure what this means though).

I'm not sure how to handle these messages but if I do that in base.py:70 : 
```
    # Otherwise, just scrub the singular value
    try:
        raw_value = base_type.parse(raw_value)
    except:
        print('Substring "0x00" not found, skipping value')
        print(raw_value)
```

It works just fine (for us), but I don't know what the consequences downstream could be if we just append the raw_value without parsing it...

I would love to make a PR to incorporate the handling of this upstream but I'm not comfortable enough without having more information and feedback.

Does anyone want to chime in on this ?
Thanks

**Comment by xmedeko** [comment]
The FIT file contains the length of data. AFAIK the parser should handle both cases - if the string has null value or does not have it at the end of the string data.

**Comment by polyvertex** [comment]
FYI this is fixed by #74 (not merged ATM)

**Comment by jeanbaptistemora** [comment]
Sorry for the response time, I've been quite swamped.

Thank you so much for your answers ! @polyvertex your solution is quite more elegant than mine, I've changed our hotfix for this one, awaiting for your PR to be merged :)

I'll close this issue.
